finality-aleph/src/network/tcp.rs

```diff
@@ -106,6 +110,15 @@ impl AddressingInformation for SignedTcpAddressingInformation {
         self.peer_id()
             .verify(&self.addressing_information.encode(), &self.signature)
     }
+
+    // Returns first address (without port) from addressing information that parses to IP address,
+    // without performing dns resolution.
+    fn lower_level_address(&self) -> Option<String> {
+        iter::once(self.addressing_information.primary_address.clone())
+            .chain(self.addressing_information.other_addresses.clone())
+            .filter_map(|address| Some(address.parse::<SocketAddr>().ok()?.ip().to_string()))
```

Comment:
Uhhhh, I thought this was wrong since we support DNS addresses, not just plain IP, but then I checked and we don't. o__o''' Well, there should probably be a task for improving that?

Reply:
Here I wanted to use the former, since as a caller of `lower_level_address` I'd expect its implementation to not use relatively expensive calls.

Reply:
Ah, you are right, it does that pretty deep in the code... Hm, could you then not filter for IP here? why are you doing that in the first place?

Reply:
Ah, I only saw the first response initially. Well, you could also assume the addresses are already correct and just return the primary one, no? It's a `String` already.

Reply:
Sure, I will do that. Previously I needed some canonical form because I was joining by IP address to get substrate's network details, but we've decided that we don't need that, so I can simplify also that.